noseiquela_orm/query.py

```diff
@@ -1,109 +1,117 @@
-from typing import Any, Dict, Tuple, Optional, Generator, TYPE_CHECKING
-from functools import partial
-from google.cloud.datastore.query import Query as GoogleQuery
+from typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
-    from .entity import Entity
+    from functools import partial
+    from typing import Any, Dict, Tuple, Optional, Generator, Iterable
+    from google.cloud.datastore.query import Query as GoogleQuery
+    from .entity import Model
 
 class QueryResult:
     def __init__(
         self,
-        query: GoogleQuery,
-        entity_instance: 'Entity',
-        limit: Optional[int]=None,
-        offset: Optional[int]=0,
-        retry: Optional[int]=None,
-        timeout: Optional[int]=None
-    ) -> None:
+        query: 'GoogleQuery',
+        entity_instance: 'Model',
+        limit: 'Optional[int]'=None,
+        offset: 'Optional[int]'=0,
+        retry: 'Optional[int]'=None,
+        timeout: 'Optional[int]'=None
+    ) -> 'None':
         self.query = query
         self.limit = limit
         self.offset = offset
         self.retry = retry
         self.timeout = timeout
         self.entity_instance = entity_instance
 
-    def first(self) -> Optional['Entity']:
-        result = list(self.query.fetch(
-            limit=1,
-            retry=self.retry,
-            timeout=self.timeout
-        ))
-        return self.entity_instance._create_from_google_entity(
-            result[0]
-        ) if result else None
```

Comment:
Moved to the "`Query`" class as that is where it seems to make the most sense to me.